lms/djangoapps/student_account/views.py

```
@@ -343,27 +343,32 @@ def get_user_orders(user):
-                                order_data = {
-                                    'number': order['number'],
-                                    'price': order['total_excl_tax'],
-                                    'title': order['lines'][0]['title'],
-                                    'order_date': strftime_localized(
-                                        date_placed.replace(tzinfo=pytz.UTC), 'SHORT_DATE'
-                                    ),
-                                    'receipt_url': EcommerceService().get_receipt_page_url(order['number'])
-                                }
+        try:
+            if order['status'].lower() == 'complete':
+                order_lines = order['lines']
+                date_placed = datetime.strptime(order['date_placed'], "%Y-%m-%dT%H:%M:%SZ")
+                order_data = {
+                    'number': order['number'],
+                    'price': order['total_excl_tax'],
+                    'order_date': strftime_localized(
+                        date_placed.replace(tzinfo=pytz.UTC), 'SHORT_DATE'
+                    ),
+                    'receipt_url': EcommerceService().get_receipt_page_url(order['number']),
+                    'lines': order_lines
+                }
+
+                if len(order_lines) == 1:
+                    product = order_lines[0].get('product')
+
+                    if product:
+                        for attribute in product['attribute_values']:
+                            if attribute['name'] == 'certificate_type' and attribute['value'] in allowed_course_modes:
                                 user_orders.append(order_data)
-                            except KeyError:
-                                log.exception('Invalid order structure: %r', order)
-                                return no_data
+                else:
+                    user_orders.append(order_data)
+        except KeyError:
```

This should go away. It will only hide real errors that I don't believe we are monitoring in Splunk.